pkg/controller/daemon/daemon_controller.go

```diff
@@ -1053,7 +1054,7 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ds *apps.DaemonSet, nodeL
 
 	var desiredNumberScheduled, currentNumberScheduled, numberMisscheduled, numberReady, updatedNumberScheduled, numberAvailable int
 	for _, node := range nodeList {
-		wantToRun, _, _, err := dsc.nodeShouldRunDaemonPod(node, ds)
+		wantToRun, _, err := dsc.nodeShouldRunDaemonPod(node, ds)
```

Comment:
wantToRun has been deleted. Changing this to shouldRun is probably fine, although the DaemonSet status will be reported a bit differently. Failure to schedule will be reported as Pending pods instead of as diff between desiredNumberSchedule and currentNumberScheduled. Is that ok?

https://github.com/kubernetes/kubernetes/blob/3760d24674d9eac92284c5d19c4a774aceab1b7f/pkg/apis/apps/types.go#L592-L594

Reply:
This PR didn't change the semantics of the current logic, what used to be named ```wantToRun``` and ```shouldSchedule``` were always set to the same value by ```nodeShouldRunDaemonPod```